service/frontend/admin_handler.go

```diff
@@ -1529,8 +1532,28 @@ func (adh *AdminHandler) ResendReplicationTasks(
 	resender := xdc.NewNDCHistoryResender(
 		adh.namespaceRegistry,
 		adh.clientBean,
-		func(ctx context.Context, request *historyservice.ReplicateEventsV2Request) error {
-			_, err1 := adh.historyClient.ReplicateEventsV2(ctx, request)
+		func(
+			ctx context.Context,
+			namespaceId namespace.ID,
+			workflowId string,
+			runId string,
+			events []*historypb.HistoryEvent,
+			versionHistory []*historyspb.VersionHistoryItem,
+		) error {
+			historyBlob, err1 := adh.eventSerializer.SerializeEvents(events, enumspb.ENCODING_TYPE_PROTO3)
+			if err1 != nil {
+				return err1
+			}
+			replicateRequest := &historyservice.ReplicateEventsV2Request{
+				NamespaceId: namespaceId.String(),
+				WorkflowExecution: &commonpb.WorkflowExecution{
+					WorkflowId: workflowId,
+					RunId:      runId,
+				},
+				Events:              historyBlob,
+				VersionHistoryItems: versionHistory,
+			}
+			_, err1 = adh.historyClient.ReplicateEventsV2(ctx, replicateRequest)
```

Comment:
Does this API invoke ReplicateHistoryEvents?

Reply:
Yeah, behind the scene, they share the same code path at history engine. While this piece of code is in frontend/admin, so we have to use RPC call to handle that. 

`ResendReplicationTasks` api seems never used in our code base. I am thinking to deprecate this API.